### ISSUE TITLE
Ignore_errors causes empty result (read_csv)

### ISSUE THREAD

**Issue by b-schmeling** [duckdb/duckdb#11296]
### What happens?

Same bug as [#10769](https://github.com/duckdb/duckdb/issues/10769) which was resolved, but it showed up again with a different file. Specifically, the header row contains 3 columns and the actual data contains 4 columns. 

Using `ignore_errors=true` in read_csv results in an empty result set. Removing `ignore_errors=true` allows the data to load. 

### To Reproduce
**Case 1**
Use file: [inconsistent_columns_3.csv](https://github.com/duckdb/duckdb/files/14705688/inconsistent_columns_3.csv)

```python
import duckdb

if __name__ == "__main__":
    res = duckdb.connect().execute(
        "SELECT * FROM read_csv('inconsistent_columns_3.csv', null_padding=true, ignore_errors=true)"
    )
    print(res.fetch_df())

```

Prints:
>Empty DataFrame
Columns: [one, two, three]
Index: []

The rejects table is also empty.

---
**Case 2**
Something similar happened with another file, but I am not exactly sure what the expected behavior of the csv reader was supposed to be for this one: [line_with_spaces.csv](https://github.com/duckdb/duckdb/files/14705874/line_with_spaces.csv)
```python
import duckdb

if __name__ == "__main__":
    res = duckdb.connect().execute(
        "SELECT * FROM read_csv('line_with_spaces.csv', null_padding=true, ignore_errors=true)"
    )
    print(res.fetch_df())
```
Print:
> Empty DataFrame
Columns: [Line with spaces]
Index: []

---
**Case 3**
Using `ignore_errors=true` for this file seemed to mess up the null_padding: [inconsistent_columns_6.csv](https://github.com/duckdb/duckdb/files/14705933/inconsistent_columns_6.csv)
```python
import duckdb
if __name__ == "__main__":
    res = duckdb.connect().execute(
        "SELECT * FROM read_csv('inconsistent_columns_6.csv', null_padding=true)"
    )
    print(res.fetch_df())
```
Prints:
>   one  two  three  four  column4
0    1    2      3     4      5.0
1    1    2      3     4      NaN
2    1    2      3     4      NaN
3    1    2      3     4      NaN

Now, adding ignore_errors=true, prints:
>    one  two  three  four
0    1    2      3     4
1    1    2      3     4
2    1    2      3     4
3    1    2      3     4
---
**Case 4**
[newline_exported.csv](https://github.com/duckdb/duckdb/files/14727486/newline_exported.csv)
Without ignore_errors:
>Date                                        text_header
0 2001-01-19                                               text
1 2001-02-19                                               text
2 2001-03-19                                             'NULL'
3 2001-04-19                                   text with, comma
4 2001-05-19                                  text with , comma
5 2001-06-19                                               NULL
6 2001-07-19                                               null
7 2001-08-19                                               None
8 2001-09-19  \ntext after the 1st new line\n\ntext after th...
9 2001-10-19                                   but we can't see

With ignore_errors, the strings on the right side disappear:
> Date       text_header
0                        1/1/19              text
1                        1/2/19              text
2                        1/3/19              NULL
3                        1/6/19              NULL
4                        1/7/19              null
5                        1/8/19              None
6                        1/9/19                 "
7   text after the 1st new line              None
8  text after the 2nd new line"              None
9                       1/10/19  but we can't see

### OS:

arm64

### DuckDB Version:

0.10.1

### DuckDB Client:

Python

### Full Name:

Bethany Schmeling

### Affiliation:

Datachat

### Have you tried this on the latest [nightly build](https://duckdb.org/docs/installation/?version=main)?

I have tested with a release build (and could not test with a nightly build)

### Have you tried the steps to reproduce? Do they include all relevant data and configuration? Does the issue you report still appear there?

- [X] Yes, I have

**Comment by cmdlineluser** [comment]
Did https://github.com/duckdb/duckdb/pull/11313 fix these?

Trying on nightly:

```python
>>> duckdb.__version__
'0.10.2-dev213'
```

```python
duckdb.sql("from read_csv('inconsistent_columns_3.csv', null_padding=true, ignore_errors=true)")
# ┌───────┬───────┬───────┬─────────┐
# │  one  │  two  │ three │ column3 │
# │ int64 │ int64 │ int64 │  int64  │
# ├───────┼───────┼───────┼─────────┤
# │     1 │     2 │     3 │       4 │
# │     1 │     2 │     3 │       4 │
# │     1 │     2 │     3 │       4 │
# │     1 │     2 │     3 │       4 │
# │     1 │     2 │     3 │       4 │
# └───────┴───────┴───────┴─────────┘
```

```python
duckdb.sql("from read_csv('line_with_spaces.csv', null_padding=true, ignore_errors=true)")
# ┌──────────────────┬─────────┬─────────┬─────────┐
# │ Line with spaces │ column1 │ column2 │ column3 │
# │     varchar      │ varchar │ varchar │ varchar │
# ├──────────────────┼─────────┼─────────┼─────────┤
# │ a                │ b       │ c       │ d       │
# │ 1                │ 2       │ 3       │ 4       │
# │ 1                │ 2       │ 3       │ 4       │
# │ 1                │ 2       │ 3       │ 4       │
# │ 1                │ 2       │ 3       │ 4       │
# └──────────────────┴─────────┴─────────┴─────────┘
```

```python
duckdb.sql("from read_csv('inconsistent_columns_6.csv', null_padding=true, ignore_errors=true)")
# ┌───────┬───────┬───────┬───────┬─────────┐
# │  one  │  two  │ three │ four  │ column4 │
# │ int64 │ int64 │ int64 │ int64 │  int64  │
# ├───────┼───────┼───────┼───────┼─────────┤
# │     1 │     2 │     3 │     4 │    NULL │
# │     1 │     2 │     3 │     4 │       5 │
# │     1 │     2 │     3 │     4 │    NULL │
# │     1 │     2 │     3 │     4 │    NULL │
# │     1 │     2 │     3 │     4 │    NULL │
# └───────┴───────┴───────┴───────┴─────────┘
```

```python
duckdb.sql("from read_csv('newline_exported.csv', null_padding=true, ignore_errors=true)")
# ┌────────────┬──────────────────────────────────────────────────────────────┐
# │    Date    │                         text_header                          │
# │    date    │                           varchar                            │
# ├────────────┼──────────────────────────────────────────────────────────────┤
# │ 2001-01-19 │ text                                                         │
# │ 2001-02-19 │ text                                                         │
# │ 2001-03-19 │ 'NULL'                                                       │
# │ 2001-04-19 │ text with, comma                                             │
# │ 2001-05-19 │ text with , comma                                            │
# │ 2001-06-19 │ NULL                                                         │
# │ 2001-07-19 │ null                                                         │
# │ 2001-08-19 │ NULL                                                         │
# │ 2001-09-19 │ \ntext after the 1st new line\n\ntext after the 2nd new line │
# │ 2001-10-19 │ but we can't see                                             │
# ├────────────┴──────────────────────────────────────────────────────────────┤
# │ 10 rows                                                         2 columns │
# └───────────────────────────────────────────────────────────────────────────┘
```

**Comment by b-schmeling** [comment]
> Did https://github.com/duckdb/duckdb/pull/11313 fix these?

This issue is fixed in my testing on `0.10.2`
